homeassistant/components/plex/media_player.py

```diff
@@ -232,52 +160,41 @@ def _clear_media_details(self):
         # Clear library Name
         self._app_name = ""
 
-    def refresh(self, device, session):
+    def update(self):
         """Refresh key device data."""
         self._clear_media_details()
 
-        if session:  # Not being triggered by Chrome or FireTablet Plex App
-            self._session = session
-        if device:
-            self._device = device
+        self._available = self.device or self.session
+
+        if self.device:
             try:
-                device_url = self._device.url("/")
+                device_url = self.device.url("/")
             except plexapi.exceptions.BadRequest:
                 device_url = "127.0.0.1"
             if "127.0.0.1" in device_url:
-                self._device.proxyThroughServer()
-            self._session = None
-            self._machine_identifier = self._device.machineIdentifier
-            self._name = NAME_FORMAT.format(self._device.title or DEVICE_DEFAULT_NAME)
-            self._device_protocol_capabilities = self._device.protocolCapabilities
-
-            # set valid session, preferring device session
-            if self._device.machineIdentifier in self.plex_sessions:
-                self._session = self.plex_sessions.get(
-                    self._device.machineIdentifier, [None, None]
-                )[0]
-
-        if self._session:
-            if (
-                self._device is not None
-                and self._device.machineIdentifier is not None
-                and self._session.players
-            ):
-                self._is_player_available = True
-                self._player = [
+                self.device.proxyThroughServer()
+            self._name = NAME_FORMAT.format(self.device.title or DEVICE_DEFAULT_NAME)
+            self._device_protocol_capabilities = self.device.protocolCapabilities
+            self._player_state = self.device.state
+
+        if not self.session:
+            self.force_idle()
+        else:
+            session_device = next(
+                (
                     p
-                    for p in self._session.players
-                    if p.machineIdentifier == self._device.machineIdentifier
-                ][0]
-                self._name = NAME_FORMAT.format(self._player.title)
-                self._player_state = self._player.state
-                self._session_username = self._session.usernames[0]
-                self._make = self._player.device
-            else:
-                self._is_player_available = False
+                    for p in self.session.players
+                    if p.machineIdentifier == self.device.machineIdentifier
+                )
+            )
+            if session_device:
```

Comment:
When is this falsy? Do we need a default in `next` call above to avoid `StopIteration` error?

Reply:
I don't think it _should_ be, but certain clients and/or the Plex API behave unexpectedly sometimes. I'll add a check and a warning just in case for now.